back/app/controllers/web_api/v1/baskets_ideas_controller.rb

```diff
@@ -52,6 +52,51 @@ def update
     end
   end
 
+  def upsert
+    # 1. Create or get basket
+    idea = Idea.find(params[:idea_id])
+    participation_context = ParticipationContextService.new.get_participation_context(idea.project)
+    participation_context_type = participation_context.phase? ? 'Phase' : 'Project'
+
+    basket = Basket.find_or_initialize_by(
+      participation_context: participation_context,
+      participation_context_type: participation_context_type,
```

Comment:
It shouldn't be necessary to specify the type. Typically, we either specify the object (`participation_context`) or we specify the type and ID.